src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -184,6 +184,13 @@ public EventTypeService(final EventTypeRepository eventTypeRepository,
             final EventType eventType = schemaEvolutionService.evolve(original, eventTypeBase);
             eventType.setDefaultStatistic(
                     validateStatisticsUpdate(original.getDefaultStatistic(), eventType.getDefaultStatistic()));
+            final Long newRetentionTime = eventTypeBase.getOptions().getRetentionTime();
+            final Long oldRetentionTime = original.getOptions().getRetentionTime();
+            if (newRetentionTime != null && !newRetentionTime.equals(oldRetentionTime)) {
```

Comment:
Actually this would cause the same problem, but the other way around. It should be possible to rollback the first operation (update topics' retention time, or update ET) if the second fails.